radix-engine/tests/integration.rs

```diff
@@ -190,6 +190,51 @@ fn test_resource_def() {
     let receipt = executor.run(transaction).unwrap();
     println!("{:?}", receipt);
     assert!(!receipt.result.is_ok());
+
+    let transaction = TransactionBuilder::new(&executor)
+        .call_function(
+            package,
+            "ResourceTest",
+            "create_fungible_wrong_resource_flags_should_fail",
+            vec![],
+            Some(account),
+        )
+        .call_method_with_all_resources(account, "deposit_batch")
+        .build(vec![key])
+        .unwrap();
+    let receipt = executor.run(transaction).unwrap();
+    println!("{:?}", receipt);
+    assert!(!receipt.result.is_ok());
+
+    let transaction = TransactionBuilder::new(&executor)
+        .call_function(
+            package,
+            "ResourceTest",
+            "create_fungible_wrong_mutable_flags_should_fail",
+            vec![],
+            Some(account),
+        )
+        .call_method_with_all_resources(account, "deposit_batch")
+        .build(vec![key])
+        .unwrap();
+    let receipt = executor.run(transaction).unwrap();
+    println!("{:?}", receipt);
+    assert!(!receipt.result.is_ok());
+
+    let transaction = TransactionBuilder::new(&executor)
+        .call_function(
+            package,
+            "ResourceTest",
+            "create_fungible_wrong_resource_permissions_should_fail",
+            vec![],
+            Some(account),
+        )
+        .call_method_with_all_resources(account, "deposit_batch")
+        .build(vec![key])
+        .unwrap();
+    let receipt = executor.run(transaction).unwrap();
+    println!("{:?}", receipt);
+    assert!(!receipt.result.is_ok());
```

Comment:
Perhaps these should be three different tests

Reply:
I agree this test is becoming a bit long. OTOH I am just adding new tests for creating resources, which is what test_resource_def does.

Reply:
I'd suggest we keep them as-is for now. At a not-too-far point in time, we need to split the `integration.rs` into multiple files, each containing a set of tests covering one aspect of the system.  This can be done as an individual task.